### PR TITLE
chore: use python lambda runtime from file

### PR DESCRIPTION
**Pull request by nitrocode**
## what

- use python lambda runtime from file

## why

- Use a pyenv's `.python-version` file to consolidate the version in a single place

Local test

```hcl
variable "runtime" {
  default = null
}

locals {
  runtime = coalesce(var.runtime, format("python%s", regex("^\\d+\\.\\d+", file("${path.module}/.python-version"))))
}
```

Terraform console with default of `null`

```bash
✗ terraform console
> var.runtime
null
> local.runtime
"python3.11"
```

```bash
✗ terraform console -var=runtime=aaaa
> local.runtime
"aaaa"
```

## references

- https://docs.aws.amazon.com/lambda/latest/dg/lambda-python.html
- https://docs.python.org/3/whatsnew/3.11.html
- https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-file-input
- https://developer.hashicorp.com/terraform/language/functions/regex

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -102,7 +102,7 @@ variable "production_workspace" {
 
 variable "runtime" {
   description = "Lambda language runtime"
-  default     = "python3.11"
+  default     = "python3.12"
```
Comment (nitrocode) [inline variables.tf]: It's possible to keep this default `null` or `""` and read the `.python-version` by default and override it if this version is supplied with something else

Comment (nitrocode) [reply variables.tf]:
```suggestion
  default     = "python3.11"
```

File: .python-version [hunk 2]
```diff
@@ -0,0 +1 @@
+3.12
```
Comment (nitrocode) [inline .python-version]:
```suggestion
3.11
```

**Review by paulschwarzenberger — COMMENTED** [review]
It's a good idea to have a single place to set the Python version.
Should it be extended to also include the Lambda Python runtime in Terraform and the version in the Lambda build script?
The version in all these different places needs to be the same.

File: variables.tf (hunk 1 above)
Comment (nitrocode) [reply variables.tf]: Addressed https://github.com/domain-protect/terraform-aws-domain-protect/pull/77/commits/5e1f5bad94a4a4e04e68bb98346748cb4cdf1952

**Review by paulschwarzenberger — APPROVED** [review]
LGTM

**Pull request merged**